test/IRGen/section_wasm.swift

```diff
@@ -0,0 +1,31 @@
+// RUN: %target-swift-frontend -enable-experimental-feature SymbolLinkageMarkers -primary-file %S/section.swift -S -parse-as-library | %FileCheck %s --check-prefix=ASM --check-prefix ASM-%target-os
+// REQUIRES: CPU=wasm32
+// REQUIRES: swift_in_compiler
+
+// Wasm cannot have user defined section name for code sections
+//
+// ASM-NOT: .section{{.*}}__TEXT,__mysection
+// ASM: .section        ".text.$s7section3fooyyF"
+// ASM: $s7section3fooyyF:
+
+// Wasm places one-time init token on .bss section after each data section
+// so need to put .section directives for each data to switch sections
```

Comment:
I find this comment a little confusing. We're declaring a pile of `text` sections, not `data` sections. Is the issue here that the WASM output adds an additional section switch after each of the variables? i.e. the WASM output looks something like

```
  .section "__TEXT,__mysection"
$s7secti0on2g0Sivp:
  .long 0
  .bss // or an equivalent .section
  // Some other stuff here
```

If so, maybe the comment could show an example of the pattern that means we need to change section every time?

Reply:
Ah, it might confuse others too, thank you.

Actually, all byte sequences except for instruction sequences are placed in "data" sections in Wasm world, even though the section name is `__TEXT, blah`. Here is a part of the Wasm output:

```asm
        .hidden $s7section2g0Sivp
        .type   $s7section2g0Sivp,@object
        .section        "__TEXT,__mysection","",@
        .globl  $s7section2g0Sivp
        .p2align        2, 0x0
$s7section2g0Sivp:
        .int32  1
        .size   $s7section2g0Sivp, 4

        .type   $s7section2g1_Wz,@object
        .section        ".bss.$s7section2g1_Wz","",@
        .p2align        2, 0x0
$s7section2g1_Wz:
        .int32  0
        .size   $s7section2g1_Wz, 4

        .hidden $s7section2g1Si_Sitvp
        .type   $s7section2g1Si_Sitvp,@object
        .section        "__TEXT,__mysection","",@
        .globl  $s7section2g1Si_Sitvp
        .p2align        2, 0x0
$s7section2g1Si_Sitvp:
        .int32  42
        .int32  43
        .size   $s7section2g1Si_Sitvp, 8

        .type   $s7section2g2_Wz,@object
        .section        ".bss.$s7section2g2_Wz","",@
        .p2align        2, 0x0
$s7section2g2_Wz:
        .int32  0
        .size   $s7section2g2_Wz, 4
```

Reply:
Actually, all byte sequence except for instruction sequences is placed in "data" section in Wasm world even though the section is named "__TEXT, blah". So all definitions except for `s7section3fooyyF` are placed in data section.

And yes, the problem here is LLVM places one-time init token and the actual data space close because they are both data sections. In other archs, token and data space are usually far because data is placed in text and token is placed in data

```
        .hidden $s7section2g0Sivp
        .type   $s7section2g0Sivp,@object
        .section        "__TEXT,__mysection","",@
        .globl  $s7section2g0Sivp
        .p2align        2, 0x0
$s7section2g0Sivp:
        .int32  1
        .size   $s7section2g0Sivp, 4

        .type   $s7section2g1_Wz,@object
        .section        ".bss.$s7section2g1_Wz","",@
        .p2align        2, 0x0
$s7section2g1_Wz:
        .int32  0
        .size   $s7section2g1_Wz, 4

        .hidden $s7section2g1Si_Sitvp
        .type   $s7section2g1Si_Sitvp,@object
        .section        "__TEXT,__mysection","",@
        .globl  $s7section2g1Si_Sitvp
        .p2align        2, 0x0
$s7section2g1Si_Sitvp:
        .int32  42
        .int32  43
        .size   $s7section2g1Si_Sitvp, 8

        .type   $s7section2g2_Wz,@object
        .section        ".bss.$s7section2g2_Wz","",@
        .p2align        2, 0x0
$s7section2g2_Wz:
        .int32  0
        .size   $s7section2g2_Wz, 4
```